cmd/crowdsec-cli/capi.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/crowdsecurity/crowdsec/pkg/apiclient"
+	"github.com/crowdsecurity/crowdsec/pkg/csconfig"
+	"github.com/crowdsecurity/crowdsec/pkg/cwhub"
+	"github.com/crowdsecurity/crowdsec/pkg/cwversion"
+	"github.com/crowdsecurity/crowdsec/pkg/models"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/go-openapi/strfmt"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"gopkg.in/yaml.v2"
+)
+
+var APIBaseURL string = "https://api.dev.crowdsec.net/"
+var APIURLPrefix string = "v2"
+
+func NewCapiCmd() *cobra.Command {
+	var cmdCapi = &cobra.Command{
+		Use:   "capi [action]",
+		Short: "Manage interraction with Central API (CAPI)",
+		Args:  cobra.MinimumNArgs(1),
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			apiclient.BaseURL, err = url.Parse(APIBaseURL)
+			if err != nil {
+				return errors.Wrapf(err, "unable to parse api url %s", APIBaseURL)
+			}
+			apiclient.URLPrefix = APIURLPrefix
+			apiclient.UserAgent = fmt.Sprintf("crowdsec/%s", cwversion.VersionStr())
+			return nil
+		},
+	}
+
+	var cmdCapiRegister = &cobra.Command{
+		Use:   "register",
+		Short: "Register to Central API (CAPI)",
+		Args:  cobra.MinimumNArgs(0),
+		Run: func(cmd *cobra.Command, args []string) {
+			Client = apiclient.NewClient(nil)
+
+			id, err := generateID()
+			if err != nil {
+				log.Fatalf("unable to generate machine id: %s", err)
+			}
+			log.Printf("your machine ID : %s", id)
+			password := strfmt.Password(generatePassword(passwordLength))
+			log.Printf("your password : %s", password)
+			_, err = Client.Auth.RegisterWatcher(context.Background(), models.WatcherRegistrationRequest{
```

Comment:
do we want to check that `resp.StatusCode == 200` ?

Reply:
It is already done by the `Do()` method